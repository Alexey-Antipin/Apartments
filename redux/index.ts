import { checkbox, toogleBox, reset } from "./reducers/checkboxReducer";
import { newsThunk, countNewsThunk } from "./reducers/newsReducer";
import { amountOfRoomsThunk } from "./reducers/mainReducer";
import { articlesThunk } from "./reducers/articlesReducer";
import { choiceCity } from "./reducers/catalogReducer";
import {
  authorizationThunk,
  accountUser,
  accountDelete,
} from "./reducers/authorizationReducer";
import {
  redistrationThunk,
  confirmDataThunk,
  toggleCaptcha,
  captchaThunk,
} from "./reducers/registrationReducer";
import {
  contactsThunk,
  closeModal,
  openModal,
} from "./reducers/contactsReducer";
import {
  selectMetroMainPage,
  selectAreaMainPage,
  selectCountRooms,
  selectCallCity,
  selectPriceMin,
  selectPriceMax,
  selectPlaces,
  defaultPrice,
  selectMetro,
  selectCity,
  selectArea,
} from "./reducers/selectReducer";

import { useAppDispatch, useAppSelector } from "../redux/hooks";
import type { RootState, AppDispatch } from "./store";

export {
  // Thunk
  authorizationThunk,
  amountOfRoomsThunk,
  redistrationThunk,
  confirmDataThunk,
  countNewsThunk,
  articlesThunk,
  contactsThunk,
  captchaThunk,
  newsThunk,

  // Method
  selectMetroMainPage,
  selectAreaMainPage,
  selectCountRooms,
  selectPriceMin,
  selectPriceMax,
  selectCallCity,
  accountDelete,
  toggleCaptcha,
  selectPlaces,
  defaultPrice,
  selectMetro,
  accountUser,
  selectCity,
  selectArea,
  choiceCity,
  closeModal,
  toogleBox,
  openModal,
  checkbox,
  reset,

  // Hooks
  useAppDispatch,
  useAppSelector,

  //Type Store
  AppDispatch,
  RootState,
};
