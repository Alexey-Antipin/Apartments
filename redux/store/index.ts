import { configureStore } from "@reduxjs/toolkit";
import authorizationSlice from "../reducers/authorizationReducer";
import registrationSlice from "../reducers/registrationReducer";
import contactsSlice from "../reducers/contactsReducer";
import newsSlice from "../reducers/newsReducer";

export const store = configureStore({
  reducer: {
    authorization: authorizationSlice,
    registration: registrationSlice,
    contacts: contactsSlice,
    news: newsSlice,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
