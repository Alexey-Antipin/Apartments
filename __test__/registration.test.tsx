import Registration from "../pages/registration";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../redux/store";

describe("Registration", () => {
  test("Head", () => {
    const { getByText } = render(
      <Provider store={store}>
        <Registration />
      </Provider>
    );
    let head = getByText(/Регистрация/i);
    expect(head).toBeInTheDocument();
  });
  test("button", () => {
    const { getByRole } = render(
      <Provider store={store}>
        <Registration />
      </Provider>
    );
    let button = getByRole("button", { name: /Зарегистрироваться/i });
    expect(button).toBeInTheDocument();
  });
  test("component snapshot", () => {
    const component = render(
      <Provider store={store}>
        <Registration />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});