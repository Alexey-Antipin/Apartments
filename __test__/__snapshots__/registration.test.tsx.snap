// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="background"
      >
        <div
          class="registration"
        >
          <form
            action="#"
            class="form"
          >
            <h1
              class="title"
            >
              Регистрация
            </h1>
            <label
              class="block"
              for="login"
            >
              <input
                class="input"
                id="login"
                name="login"
                placeholder="Логин"
                type="text"
                value=""
              />
              <div
                class="icon"
              >
                <svg
                  fill="#686868"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    clip-path="url(#clip0_305_638)"
                  >
                    <path
                      d="M10.0013 0C7.14418 0 4.80859 2.33559 4.80859 5.19275C4.80859 8.04991 7.14418 10.3855 10.0013 10.3855C12.8585 10.3855 15.1941 8.04991 15.1941 5.19275C15.1941 2.33559 12.8585 0 10.0013 0Z"
                    />
                    <path
                      d="M18.913 14.535C18.7769 14.1949 18.5955 13.8774 18.3915 13.5826C17.3484 12.0407 15.7384 11.0203 13.9244 10.7708C13.6976 10.7482 13.4482 10.7935 13.2668 10.9296C12.3144 11.6325 11.1806 11.9953 10.0014 11.9953C8.82228 11.9953 7.68851 11.6325 6.73612 10.9296C6.5547 10.7935 6.30526 10.7255 6.07853 10.7708C4.26446 11.0203 2.63183 12.0407 1.61143 13.5826C1.40735 13.8774 1.22592 14.2176 1.0899 14.535C1.02189 14.6711 1.04454 14.8298 1.11256 14.9659C1.29398 15.2833 1.52071 15.6008 1.72479 15.8729C2.04224 16.3038 2.38239 16.6892 2.76789 17.052C3.08534 17.3695 3.44815 17.6643 3.81099 17.9591C5.60236 19.297 7.75657 19.9999 9.97879 19.9999C12.201 19.9999 14.3552 19.2969 16.1466 17.9591C16.5094 17.687 16.8722 17.3695 17.1897 17.052C17.5525 16.6892 17.9153 16.3037 18.2328 15.8729C18.4595 15.5781 18.6636 15.2833 18.845 14.9659C18.9583 14.8298 18.981 14.6711 18.913 14.535Z"
                    />
                  </g>
                  <defs>
                    <clippath
                      id="clip0_305_638"
                    >
                      <rect
                        fill="white"
                        height="20"
                        width="20"
                      />
                    </clippath>
                  </defs>
                </svg>
              </div>
            </label>
            <label
              class="block"
              for="email"
            >
              <input
                class="input"
                id="email"
                name="email"
                placeholder="Электронная почта"
                type="email"
                value=""
              />
              <div
                class="icon"
              >
                <svg
                  fill="#686868"
                  height="18"
                  viewBox="0 0 18 18"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    clip-path="url(#clip0_305_651)"
                  >
                    <path
                      d="M15.937 15.6252C16.3304 15.6252 16.6713 15.4953 16.961 15.2389L12.003 10.2806C11.8841 10.3658 11.7688 10.4486 11.6598 10.5274C11.2888 10.8008 10.9877 11.0141 10.7564 11.167C10.5252 11.3202 10.2176 11.4763 9.8336 11.6357C9.44935 11.7954 9.09137 11.8749 8.75928 11.8749H8.74956H8.73984C8.40773 11.8749 8.04975 11.7954 7.66552 11.6357C7.2813 11.4763 6.97368 11.3202 6.7427 11.167C6.51149 11.0141 6.21051 10.8008 5.83929 10.5274C5.73584 10.4516 5.62111 10.3684 5.49707 10.2793L0.538086 15.2389C0.827817 15.4953 1.16889 15.6252 1.56223 15.6252H15.937Z"
                    />
                    <path
                      d="M0.986433 7.15831C0.615429 6.91096 0.286427 6.62768 0 6.30859V13.8521L4.36999 9.48213C3.49574 8.87179 2.36932 8.09807 0.986433 7.15831Z"
                    />
                    <path
                      d="M16.524 7.15831C15.1939 8.05861 14.0633 8.83366 13.1323 9.48388L17.5005 13.8523V6.30859C17.2205 6.62126 16.895 6.90436 16.524 7.15831Z"
                    />
                    <path
                      d="M15.9377 1.875H1.56289C1.0614 1.875 0.675839 2.04435 0.405743 2.38269C0.135425 2.72119 0.000488281 3.14452 0.000488281 3.65221C0.000488281 4.0623 0.179557 4.50663 0.537535 4.98535C0.895513 5.46387 1.27643 5.83973 1.6801 6.11313C1.90139 6.26948 2.56874 6.73342 3.68213 7.50479C4.28317 7.92129 4.80585 8.28432 5.25502 8.5976C5.63787 8.86436 5.96805 9.09537 6.24066 9.28709C6.27195 9.30904 6.32117 9.34424 6.38649 9.39095C6.45687 9.4415 6.54592 9.50565 6.65579 9.58498C6.86736 9.738 7.04313 9.86169 7.18312 9.95618C7.32291 10.0507 7.49229 10.1563 7.691 10.2735C7.88954 10.3906 8.07681 10.4786 8.25257 10.5372C8.42837 10.5957 8.5911 10.6251 8.74081 10.6251H8.75054H8.76026C8.90994 10.6251 9.07271 10.5957 9.24854 10.5372C9.42427 10.4786 9.61137 10.3908 9.81008 10.2735C10.0086 10.1563 10.1777 10.0505 10.318 9.95618C10.458 9.86169 10.6337 9.73803 10.8453 9.58498C10.955 9.50565 11.044 9.44147 11.1144 9.39111C11.1798 9.34421 11.229 9.30923 11.2604 9.28709C11.4728 9.13931 11.8038 8.90925 12.2487 8.6003C13.0583 8.03777 14.2507 7.20982 15.8308 6.11313C16.306 5.78121 16.703 5.38066 17.0221 4.91205C17.3406 4.44344 17.5003 3.95189 17.5003 3.43756C17.5003 3.00784 17.3455 2.64014 17.0365 2.33389C16.7272 2.02802 16.3609 1.875 15.9377 1.875Z"
                    />
                  </g>
                  <defs>
                    <clippath
                      id="clip0_305_651"
                    >
                      <rect
                        fill="white"
                        height="17.5"
                        width="17.5"
                      />
                    </clippath>
                  </defs>
                </svg>
              </div>
            </label>
            <label
              class="block"
              for="password"
            >
              <input
                class="input"
                id="password"
                name="password"
                placeholder="Пароль"
                type="password"
                value=""
              />
              <div
                class="icon"
              >
                <svg
                  fill="#686868"
                  height="20"
                  viewBox="0 0 16 20"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.625 7.5H13V5C13 2.2425 10.7575 0 7.99999 0C5.24249 0 2.99999 2.2425 2.99999 5V7.5H2.37499C1.34166 7.5 0.499992 8.34083 0.499992 9.375V18.125C0.499992 19.1592 1.34166 20 2.37499 20H13.625C14.6583 20 15.5 19.1592 15.5 18.125V9.375C15.5 8.34083 14.6583 7.5 13.625 7.5ZM4.66666 5C4.66666 3.16167 6.16166 1.66667 7.99999 1.66667C9.83833 1.66667 11.3333 3.16167 11.3333 5V7.5H4.66666V5ZM8.83333 13.935V15.8333C8.83333 16.2933 8.46083 16.6667 7.99999 16.6667C7.53916 16.6667 7.16666 16.2933 7.16666 15.8333V13.935C6.67083 13.6458 6.33333 13.1142 6.33333 12.5C6.33333 11.5808 7.08083 10.8333 7.99999 10.8333C8.91916 10.8333 9.66666 11.5808 9.66666 12.5C9.66666 13.1142 9.32916 13.6458 8.83333 13.935Z"
                  />
                </svg>
              </div>
            </label>
            <label
              class="block"
              for="confirmPassword"
            >
              <input
                class="input"
                id="confirmPassword"
                name="confirmPassword"
                placeholder="Повторите пароль"
                type="password"
                value=""
              />
              <div
                class="icon"
              >
                <svg
                  fill="#686868"
                  height="20"
                  viewBox="0 0 16 20"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.625 7.5H13V5C13 2.2425 10.7575 0 7.99999 0C5.24249 0 2.99999 2.2425 2.99999 5V7.5H2.37499C1.34166 7.5 0.499992 8.34083 0.499992 9.375V18.125C0.499992 19.1592 1.34166 20 2.37499 20H13.625C14.6583 20 15.5 19.1592 15.5 18.125V9.375C15.5 8.34083 14.6583 7.5 13.625 7.5ZM4.66666 5C4.66666 3.16167 6.16166 1.66667 7.99999 1.66667C9.83833 1.66667 11.3333 3.16167 11.3333 5V7.5H4.66666V5ZM8.83333 13.935V15.8333C8.83333 16.2933 8.46083 16.6667 7.99999 16.6667C7.53916 16.6667 7.16666 16.2933 7.16666 15.8333V13.935C6.67083 13.6458 6.33333 13.1142 6.33333 12.5C6.33333 11.5808 7.08083 10.8333 7.99999 10.8333C8.91916 10.8333 9.66666 11.5808 9.66666 12.5C9.66666 13.1142 9.32916 13.6458 8.83333 13.935Z"
                  />
                </svg>
              </div>
            </label>
            <div
              class="captcha"
            >
              <label
                class="label"
                for="captcha"
              >
                <div
                  class="size-toggle"
                >
                  <button
                    class="button"
                    disabled=""
                    id="captcha"
                    type="button"
                  />
                  <div
                    class="button-spinner"
                    id="spinner"
                  >
                    <svg
                      data-name="Isolation Mode"
                      fill="#84b0ff"
                      height="24"
                      id="Isolation_Mode"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M.415,12.051a12,12,0,0,1,23.17-4.364L20.791,8.779a9,9,0,1,0,0,6.545l2.794,1.092A12,12,0,0,1,.415,12.051Z"
                      />
                    </svg>
                  </div>
                  <div
                    class="mark"
                    id="mark"
                  >
                    <svg
                      fill="none"
                      height="10"
                      viewBox="0 0 14 10"
                      width="14"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0.5 5.41478L4.95177 9.87505L13.5 1.33527L12.2728 0.125L4.95177 7.43749L1.71024 4.19599L0.5 5.41478Z"
                        fill="green"
                      />
                    </svg>
                  </div>
                  <div
                    class="checkbox"
                    id="checkbox"
                  />
                </div>
                <p
                  class="text"
                >
                  Я не робот
                </p>
              </label>
              <div
                class="block-right"
              >
                <label
                  class="image"
                  for="captcha"
                >
                  <img
                    alt="captcha"
                    data-nimg="1"
                    decoding="async"
                    height="30"
                    loading="lazy"
                    src="/_next/image?url=%2Fcaptcha%2Fcaptcha.png&w=64&q=75"
                    srcset="/_next/image?url=%2Fcaptcha%2Fcaptcha.png&w=32&q=75 1x, /_next/image?url=%2Fcaptcha%2Fcaptcha.png&w=64&q=75 2x"
                    style="color: transparent;"
                    width="30"
                  />
                  <p
                    class="text-image"
                  >
                    reCAPTCHA
                  </p>
                </label>
                <div
                  class="block-link"
                >
                  <a
                    class="text-under"
                    href="./"
                  >
                    Конфиденциальность
                  </a>
                  <div
                    class="text-under"
                  >
                    -
                  </div>
                  <a
                    class="text-under"
                    href="./"
                  >
                    Условия использования
                  </a>
                </div>
              </div>
            </div>
            <button
              class="send"
              disabled=""
              type="submit"
            >
              Зарегистрироваться
            </button>
          </form>
          <div
            class="block-text"
          >
            <p
              class="text-title"
            >
              Пользователь обязуется:
            </p>
            <ul>
              <li
                class="text-paragraph"
              >
                <span
                  class="text-round"
                />
                предоставлять достоверную и актуальную 
      информацию при регистрации и добавлении объекта;
              </li>
              <li
                class="text-paragraph"
              >
                <span
                  class="text-round"
                />
                добавлять фотографии объектов соответствующие
    действительности. Администрация сайта sdaem.by оставляет за
    собой право удалять любую информацию, размещенную
    пользователем, если сочтет, что информация не соответствует
    действительности, носит оскорбительный характер, нарушает
    права и законные интересы других граждан либо действующее 
    законодательство Республики Беларусь.
              </li>
            </ul>
            <p
              class="text-account"
            >
              Уже есть аккаунт?
              <a
                class="text-entrance"
                href="/authorization"
              >
                Войдите
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="background"
    >
      <div
        class="registration"
      >
        <form
          action="#"
          class="form"
        >
          <h1
            class="title"
          >
            Регистрация
          </h1>
          <label
            class="block"
            for="login"
          >
            <input
              class="input"
              id="login"
              name="login"
              placeholder="Логин"
              type="text"
              value=""
            />
            <div
              class="icon"
            >
              <svg
                fill="#686868"
                height="20"
                viewBox="0 0 20 20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#clip0_305_638)"
                >
                  <path
                    d="M10.0013 0C7.14418 0 4.80859 2.33559 4.80859 5.19275C4.80859 8.04991 7.14418 10.3855 10.0013 10.3855C12.8585 10.3855 15.1941 8.04991 15.1941 5.19275C15.1941 2.33559 12.8585 0 10.0013 0Z"
                  />
                  <path
                    d="M18.913 14.535C18.7769 14.1949 18.5955 13.8774 18.3915 13.5826C17.3484 12.0407 15.7384 11.0203 13.9244 10.7708C13.6976 10.7482 13.4482 10.7935 13.2668 10.9296C12.3144 11.6325 11.1806 11.9953 10.0014 11.9953C8.82228 11.9953 7.68851 11.6325 6.73612 10.9296C6.5547 10.7935 6.30526 10.7255 6.07853 10.7708C4.26446 11.0203 2.63183 12.0407 1.61143 13.5826C1.40735 13.8774 1.22592 14.2176 1.0899 14.535C1.02189 14.6711 1.04454 14.8298 1.11256 14.9659C1.29398 15.2833 1.52071 15.6008 1.72479 15.8729C2.04224 16.3038 2.38239 16.6892 2.76789 17.052C3.08534 17.3695 3.44815 17.6643 3.81099 17.9591C5.60236 19.297 7.75657 19.9999 9.97879 19.9999C12.201 19.9999 14.3552 19.2969 16.1466 17.9591C16.5094 17.687 16.8722 17.3695 17.1897 17.052C17.5525 16.6892 17.9153 16.3037 18.2328 15.8729C18.4595 15.5781 18.6636 15.2833 18.845 14.9659C18.9583 14.8298 18.981 14.6711 18.913 14.535Z"
                  />
                </g>
                <defs>
                  <clippath
                    id="clip0_305_638"
                  >
                    <rect
                      fill="white"
                      height="20"
                      width="20"
                    />
                  </clippath>
                </defs>
              </svg>
            </div>
          </label>
          <label
            class="block"
            for="email"
          >
            <input
              class="input"
              id="email"
              name="email"
              placeholder="Электронная почта"
              type="email"
              value=""
            />
            <div
              class="icon"
            >
              <svg
                fill="#686868"
                height="18"
                viewBox="0 0 18 18"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#clip0_305_651)"
                >
                  <path
                    d="M15.937 15.6252C16.3304 15.6252 16.6713 15.4953 16.961 15.2389L12.003 10.2806C11.8841 10.3658 11.7688 10.4486 11.6598 10.5274C11.2888 10.8008 10.9877 11.0141 10.7564 11.167C10.5252 11.3202 10.2176 11.4763 9.8336 11.6357C9.44935 11.7954 9.09137 11.8749 8.75928 11.8749H8.74956H8.73984C8.40773 11.8749 8.04975 11.7954 7.66552 11.6357C7.2813 11.4763 6.97368 11.3202 6.7427 11.167C6.51149 11.0141 6.21051 10.8008 5.83929 10.5274C5.73584 10.4516 5.62111 10.3684 5.49707 10.2793L0.538086 15.2389C0.827817 15.4953 1.16889 15.6252 1.56223 15.6252H15.937Z"
                  />
                  <path
                    d="M0.986433 7.15831C0.615429 6.91096 0.286427 6.62768 0 6.30859V13.8521L4.36999 9.48213C3.49574 8.87179 2.36932 8.09807 0.986433 7.15831Z"
                  />
                  <path
                    d="M16.524 7.15831C15.1939 8.05861 14.0633 8.83366 13.1323 9.48388L17.5005 13.8523V6.30859C17.2205 6.62126 16.895 6.90436 16.524 7.15831Z"
                  />
                  <path
                    d="M15.9377 1.875H1.56289C1.0614 1.875 0.675839 2.04435 0.405743 2.38269C0.135425 2.72119 0.000488281 3.14452 0.000488281 3.65221C0.000488281 4.0623 0.179557 4.50663 0.537535 4.98535C0.895513 5.46387 1.27643 5.83973 1.6801 6.11313C1.90139 6.26948 2.56874 6.73342 3.68213 7.50479C4.28317 7.92129 4.80585 8.28432 5.25502 8.5976C5.63787 8.86436 5.96805 9.09537 6.24066 9.28709C6.27195 9.30904 6.32117 9.34424 6.38649 9.39095C6.45687 9.4415 6.54592 9.50565 6.65579 9.58498C6.86736 9.738 7.04313 9.86169 7.18312 9.95618C7.32291 10.0507 7.49229 10.1563 7.691 10.2735C7.88954 10.3906 8.07681 10.4786 8.25257 10.5372C8.42837 10.5957 8.5911 10.6251 8.74081 10.6251H8.75054H8.76026C8.90994 10.6251 9.07271 10.5957 9.24854 10.5372C9.42427 10.4786 9.61137 10.3908 9.81008 10.2735C10.0086 10.1563 10.1777 10.0505 10.318 9.95618C10.458 9.86169 10.6337 9.73803 10.8453 9.58498C10.955 9.50565 11.044 9.44147 11.1144 9.39111C11.1798 9.34421 11.229 9.30923 11.2604 9.28709C11.4728 9.13931 11.8038 8.90925 12.2487 8.6003C13.0583 8.03777 14.2507 7.20982 15.8308 6.11313C16.306 5.78121 16.703 5.38066 17.0221 4.91205C17.3406 4.44344 17.5003 3.95189 17.5003 3.43756C17.5003 3.00784 17.3455 2.64014 17.0365 2.33389C16.7272 2.02802 16.3609 1.875 15.9377 1.875Z"
                  />
                </g>
                <defs>
                  <clippath
                    id="clip0_305_651"
                  >
                    <rect
                      fill="white"
                      height="17.5"
                      width="17.5"
                    />
                  </clippath>
                </defs>
              </svg>
            </div>
          </label>
          <label
            class="block"
            for="password"
          >
            <input
              class="input"
              id="password"
              name="password"
              placeholder="Пароль"
              type="password"
              value=""
            />
            <div
              class="icon"
            >
              <svg
                fill="#686868"
                height="20"
                viewBox="0 0 16 20"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.625 7.5H13V5C13 2.2425 10.7575 0 7.99999 0C5.24249 0 2.99999 2.2425 2.99999 5V7.5H2.37499C1.34166 7.5 0.499992 8.34083 0.499992 9.375V18.125C0.499992 19.1592 1.34166 20 2.37499 20H13.625C14.6583 20 15.5 19.1592 15.5 18.125V9.375C15.5 8.34083 14.6583 7.5 13.625 7.5ZM4.66666 5C4.66666 3.16167 6.16166 1.66667 7.99999 1.66667C9.83833 1.66667 11.3333 3.16167 11.3333 5V7.5H4.66666V5ZM8.83333 13.935V15.8333C8.83333 16.2933 8.46083 16.6667 7.99999 16.6667C7.53916 16.6667 7.16666 16.2933 7.16666 15.8333V13.935C6.67083 13.6458 6.33333 13.1142 6.33333 12.5C6.33333 11.5808 7.08083 10.8333 7.99999 10.8333C8.91916 10.8333 9.66666 11.5808 9.66666 12.5C9.66666 13.1142 9.32916 13.6458 8.83333 13.935Z"
                />
              </svg>
            </div>
          </label>
          <label
            class="block"
            for="confirmPassword"
          >
            <input
              class="input"
              id="confirmPassword"
              name="confirmPassword"
              placeholder="Повторите пароль"
              type="password"
              value=""
            />
            <div
              class="icon"
            >
              <svg
                fill="#686868"
                height="20"
                viewBox="0 0 16 20"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.625 7.5H13V5C13 2.2425 10.7575 0 7.99999 0C5.24249 0 2.99999 2.2425 2.99999 5V7.5H2.37499C1.34166 7.5 0.499992 8.34083 0.499992 9.375V18.125C0.499992 19.1592 1.34166 20 2.37499 20H13.625C14.6583 20 15.5 19.1592 15.5 18.125V9.375C15.5 8.34083 14.6583 7.5 13.625 7.5ZM4.66666 5C4.66666 3.16167 6.16166 1.66667 7.99999 1.66667C9.83833 1.66667 11.3333 3.16167 11.3333 5V7.5H4.66666V5ZM8.83333 13.935V15.8333C8.83333 16.2933 8.46083 16.6667 7.99999 16.6667C7.53916 16.6667 7.16666 16.2933 7.16666 15.8333V13.935C6.67083 13.6458 6.33333 13.1142 6.33333 12.5C6.33333 11.5808 7.08083 10.8333 7.99999 10.8333C8.91916 10.8333 9.66666 11.5808 9.66666 12.5C9.66666 13.1142 9.32916 13.6458 8.83333 13.935Z"
                />
              </svg>
            </div>
          </label>
          <div
            class="captcha"
          >
            <label
              class="label"
              for="captcha"
            >
              <div
                class="size-toggle"
              >
                <button
                  class="button"
                  disabled=""
                  id="captcha"
                  type="button"
                />
                <div
                  class="button-spinner"
                  id="spinner"
                >
                  <svg
                    data-name="Isolation Mode"
                    fill="#84b0ff"
                    height="24"
                    id="Isolation_Mode"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M.415,12.051a12,12,0,0,1,23.17-4.364L20.791,8.779a9,9,0,1,0,0,6.545l2.794,1.092A12,12,0,0,1,.415,12.051Z"
                    />
                  </svg>
                </div>
                <div
                  class="mark"
                  id="mark"
                >
                  <svg
                    fill="none"
                    height="10"
                    viewBox="0 0 14 10"
                    width="14"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0.5 5.41478L4.95177 9.87505L13.5 1.33527L12.2728 0.125L4.95177 7.43749L1.71024 4.19599L0.5 5.41478Z"
                      fill="green"
                    />
                  </svg>
                </div>
                <div
                  class="checkbox"
                  id="checkbox"
                />
              </div>
              <p
                class="text"
              >
                Я не робот
              </p>
            </label>
            <div
              class="block-right"
            >
              <label
                class="image"
                for="captcha"
              >
                <img
                  alt="captcha"
                  data-nimg="1"
                  decoding="async"
                  height="30"
                  loading="lazy"
                  src="/_next/image?url=%2Fcaptcha%2Fcaptcha.png&w=64&q=75"
                  srcset="/_next/image?url=%2Fcaptcha%2Fcaptcha.png&w=32&q=75 1x, /_next/image?url=%2Fcaptcha%2Fcaptcha.png&w=64&q=75 2x"
                  style="color: transparent;"
                  width="30"
                />
                <p
                  class="text-image"
                >
                  reCAPTCHA
                </p>
              </label>
              <div
                class="block-link"
              >
                <a
                  class="text-under"
                  href="./"
                >
                  Конфиденциальность
                </a>
                <div
                  class="text-under"
                >
                  -
                </div>
                <a
                  class="text-under"
                  href="./"
                >
                  Условия использования
                </a>
              </div>
            </div>
          </div>
          <button
            class="send"
            disabled=""
            type="submit"
          >
            Зарегистрироваться
          </button>
        </form>
        <div
          class="block-text"
        >
          <p
            class="text-title"
          >
            Пользователь обязуется:
          </p>
          <ul>
            <li
              class="text-paragraph"
            >
              <span
                class="text-round"
              />
              предоставлять достоверную и актуальную 
      информацию при регистрации и добавлении объекта;
            </li>
            <li
              class="text-paragraph"
            >
              <span
                class="text-round"
              />
              добавлять фотографии объектов соответствующие
    действительности. Администрация сайта sdaem.by оставляет за
    собой право удалять любую информацию, размещенную
    пользователем, если сочтет, что информация не соответствует
    действительности, носит оскорбительный характер, нарушает
    права и законные интересы других граждан либо действующее 
    законодательство Республики Беларусь.
            </li>
          </ul>
          <p
            class="text-account"
          >
            Уже есть аккаунт?
            <a
              class="text-entrance"
              href="/authorization"
            >
              Войдите
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
